lib/DB/Exception.pm

```diff
@@ -101,13 +101,17 @@ use Exception::Class (
 
 DB::Exception::UndefinedCourseField->Trace(1);
 DB::Exception::InvalidCourseField->Trace(1);
+DB::Exception::UserSetNotInCourse->Trace(1);
+DB::Exception::SetNotInCourse->Trace(1);
 # DB::Exception::UserNotInCourse->Trace(1);
 DB::Exception::UserNotFound->Trace(1);
-# DB::Exception::CourseNotFound->Trace(1);
+DB::Exception::CourseExists->Trace(1);
```

Comment:
The point of `CourseExists` (and `UserSetExists`, etc.) is if you are trying to create a course that already exists, then you get this exception thrown.  I have both because they are useful in different situations. 

The lines in Exception from line 102 down is to set Trace.  I had commented some out because perlcritic yells about this at some level (but I don't think at level 4 where we are using it).

Reply:
In that case, the exception should be `CourseAlreadyExists`.  An exception by the name of `CourseExists` doesn't make sense.  That is the point.  You have the exception `SetAlreadyExists`, so be consistent with that naming.